go/interpreter/lfvm/memory.go

```diff
@@ -170,60 +178,43 @@ func (m *Memory) SetWord(offset uint64, value *uint256.Int, c *context) error {
 	return nil
 }
 
-func (m *Memory) Set(offset, size uint64, value []byte) error {
+func (m *Memory) set(offset, size uint64, value []byte) error {
 	if size > 0 {
 		if offset+size < offset {
 			return errGasUintOverflow
 		}
-		if offset+size > m.Len() {
-			return fmt.Errorf("memory too small, size %d, attempted to write %d bytes at %d", m.Len(), size, offset)
+		if offset+size > m.len() {
+			return makeInsufficientMemoryError(m.len(), size, offset)
 		}
 		copy(m.store[offset:offset+size], value)
 	}
 	return nil
 }
 
-func (m *Memory) SetWithCapacityAndGasCheck(offset, size uint64, value []byte, c *context) error {
-	err := m.EnsureCapacity(offset, size, c)
-	if err != nil {
-		return err
-	}
-	err = m.Set(offset, size, value)
-	if err != nil {
-		return err
-	}
-	return nil
+func makeInsufficientMemoryError(memSize, size, offset uint64) error {
+	return tosca.ConstError(fmt.Sprintf("memory too small, size %d, attempted to write %d bytes at %d", memSize, size, offset))
 }
 
-func (m *Memory) SetWithCapacityCheck(offset, size uint64, value []byte) error {
-	m.EnsureCapacityWithoutGas(size)
-	if size > 0 {
-		if offset+size < offset {
-			return errGasUintOverflow
-		}
-		if offset+size > m.Len() {
-			return fmt.Errorf("memory too small, size %d, attempted to write %d bytes at %d", m.Len(), size, offset)
-		}
-		copy(m.store[offset:offset+size], value)
+func (m *Memory) setWithCapacityAndGasCheck(offset, size uint64, value []byte, c *context) error {
+	err := m.expandMemoryAndCharge(offset, size, c)
+	if err != nil {
+		return err
 	}
-	return nil
+	return m.set(offset, size, value)
 }
 
-func (m *Memory) CopyWord(offset uint64, trg *uint256.Int, c *context) error {
-	err := m.EnsureCapacity(offset, 32, c)
+func (m *Memory) getWord(offset uint64, trg *uint256.Int, c *context) error {
```

Comment:
Should this return a `uint256.Int` instead (a value, not a pointer) since it is a `get`? Otherwise maybe call it `readWord`?